src/Snowflake/Command.php

```diff
@@ -149,18 +226,270 @@ public static function cloneDatabaseFromShared(Connection $connection, array $da
                         QueryBuilder::quoteIdentifier($schemaName),
                         QueryBuilder::quoteIdentifier($tableName),
                     ));
-                }
 
-                var_dump($schemaName);
-                if ($k > 9) {
-                    exit();
+                    foreach ($tableGrants as $tableGrant) {
+                        self::assignGrantToRole($connection, $tableGrant);
+                    }
                 }
             }
         }
     }
 
-    public static function exportUsersAndRoles(Connection $sourceSnflkConnection): array
+    public static function createRole(Connection $connection, array $role): void
+    {
+        if (isset($role['granted_by'])) {
+            self::useRole($connection, $role['granted_by']);
+        }
+
+        $connection->query(sprintf(
+            'CREATE ROLE %s',
+            $role['name']
+        ));
+
+        self::grantRoleToUser(
+            $connection,
+            (string) getenv('SNOWFLAKE_DESTINATION_ACCOUNT_USERNAME'),
```

Comment:
Fakt je role určitě vždycky grantnutá tomu root usernamu? To bych si myslel, že možná ne

Reply:
Grantuju si vytvořenou roli na migračního uživatele abych se na ty role mohl přepínat.... po smazání uživatele pro migraci tyhle granty zmizí

Reply:
Jasně :) Sorry, myslel jsem, že tohle je ta "destination" a ne "migration"...

Reply:
no je to "destination" ale mám usera který je ACCOUNTADMIN na migraci, kterého bych po celým procesu smazal a abych se mohl přepínat mezi rolema a zakládat resourcy a nemusel se přihlašovat pod jiným userem kterému fakticky role má patřit tak mu nagrantuju tu roli